### DISCUSSION TITLE
How is Postgres scanned and where is the data stored?

### DISCUSSION THREAD

**Discussion by rinormaloku** [duckdb/duckdb#7556] [General]
I'm planning to use DuckDb as a long-running process to draw analytics out of my PostgreSQL database, and expose those through an API to clients.

Lacking information on how DuckDb works, I have a few questions that can help me understand if that is the right use case:
- Using the [postgres scanner](https://duckdb.org/docs/extensions/postgres_scanner.html) is the scanned data stored in memory or locally as a file or we are constantly scanning Postgres?
- - If the latter is true, then why use DuckDB instead of using postgres only?

Maybe just explaining how the DuckDB works would help me answer a lot of those questions. Is my understanding correct that it keeps a local copy (or an in-memory copy) of the data in a columnar format and then it is fast to query the information?

**Comment by arjenpdevries** [comment]
Processing analytics queries over large datasets may be faster than in postgres, that would be the main motivation.

**Comment by lnkuiper** [comment]
Our [blog](https://duckdb.org/2022/09/30/postgres-scanner.html) shows that DuckDB is significantly faster on specific kinds of queries, even when reading from PG directly.

The data is stored in PG, DuckDB just reads the data.

Another reason to use DuckDB for this is that DuckDB is able to combine multiple data source easily, e.g., joining PG tables with csv/parquet/json/etc.
